airflow/decorators/base.py

```diff
@@ -315,7 +315,9 @@ def __call__(self, *args: FParams.args, **kwargs: FParams.kwargs) -> XComArg:
             multiple_outputs=self.multiple_outputs,
             **self.kwargs,
         )
-        if self.function.__doc__:
+        op_doc_attrs = [op.doc, op.doc_json, op.doc_md, op.doc_rst, op.doc_yaml]
+        # Set the task's doc_md to the function's docstring if it exists and no other doc* args are set.
+        if self.function.__doc__ and not any(op_doc_attrs):
```

Comment:
Checking against _all_ of the doc attrs for tasks rather than just `doc_md`. Thinking users most likely wouldn't set more than one of these at a time.